Ide

rmdijkman opened the pull request:
Fully redone the visualization as a combination of PyQT and PyGame. The resulting library can be called in exactly the same manner as before, but has an improved look-and-feel.

Moreover, the base now also has its own main method, which allows running the simulator as an application that can load BPMN models for simulation.

AdamBanham commented:
We should probably have working tests if you commit this into main. 

Judging from the tests, we need to update pyproject.toml so that the package builds with the correct dependencies. Otherwise installers from pypi wont work.

Added a patch for the dependencies.

rmdijkman commented:
Thanks for the review and for fixing the online checks, Adam. I updated the documentation and added a description of the main setup of the visualisation module and the event handling system. The main documentation is in the __init__ file of the module (as according to co-pilot it is the convention to put it there).

I fixed another minor bug and all online checks now seem to pass.

I gathered that you still wanted to run one more check, but did not fully understand it. In any case, let me know if you are ready for the merge.

AdamBanham commented:
I was just mentioning that I would play around with making some modules using the new setup once merged with the main line. Unsure if the old hooks still work in the more unified setup (or if they still make sense). 

If you are happy, let's merge and start playing around with the new bits.

I will say that the new loop feels a bit more sluggish even at max speed for my liking, but that may be being used to my other project at the moment rather than anything here.
